release_notes.md

```diff
@@ -0,0 +1,4 @@
+# What's Changed
```

Comment:
please remove this file since we are tracking release notes here https://github.com/aws-observability/aws-otel-swift/releases/tag/1.0.0